Samples/NetworkMediaStreamer/CameraRTPStreamerApp/CameraRTPStreamerApp.cpp

```diff
@@ -0,0 +1,419 @@
+        hr = spSinkWriter->SetInputMediaType(0, spInType.get(), nullptr);
+        bEnableHWTransforms = !bEnableHWTransforms;
+    } while (hr == MF_E_TOPO_CODEC_NOT_FOUND);
+    check_hresult(hr);
+    check_hresult(spSinkWriter->BeginWriting());
+
+    return spSinkWriter.detach();
+}
+#endif
+
+PROCESS_INFORMATION StartLoggerConsole(std::wstring filename)
+{
+        STARTUPINFO si,si1;
+        PROCESS_INFORMATION pi;
+        GetStartupInfo(&si1);
+        ZeroMemory(&si, sizeof(si));
+        si.cb = sizeof(si);
+        ZeroMemory(&pi, sizeof(pi));
+        std::wstring cmdline = L"powershell Get-Content " + filename + L" -Wait ";
+        std::wstring title = L"RTSPServer Log Console";
+        si.lpTitle = (LPWSTR)title.c_str();
+        
+        si.dwX = si1.dwX + 500;
+        si.dwY = si1.dwY;
+        si.dwXCountChars = 40;
+        si.dwYCountChars = 10;
+        si.dwFlags = STARTF_USESHOWWINDOW | STARTF_USEPOSITION | STARTF_PREVENTPINNING | STARTF_USECOUNTCHARS;
+        si.wShowWindow = SW_SHOWNOACTIVATE;// | SW_HIDE;
+        // Start the child process. 
+        if (!CreateProcess(NULL,   // No module name (use command line)
+            (LPWSTR)cmdline.c_str(),        // Command line
+            NULL,           // Process handle not inheritable
+            NULL,           // Thread handle not inheritable
+            FALSE,          // Set handle inheritance to FALSE
+            CREATE_NEW_CONSOLE,              // No creation flags
+            NULL,           // Use parent's environment block
+            NULL,           // Use parent's starting directory 
+            &si,            // Pointer to STARTUPINFO structure
+            &pi)           // Pointer to PROCESS_INFORMATION structure
+            )
+        {
+            check_win32(GetLastError());
+        }
+        return pi;
+    
+}
+
+void StopLoggerConsole(PROCESS_INFORMATION pi)
+{
+    TerminateProcess(pi.hProcess, 0);
+    // Close process and thread handles. 
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+}
+int main()
+{
+    std::cout << "VideoSteamer App \n";
+    PROCESS_INFORMATION loggerConsole;
+    try
+    {
+        auto filteredDevices = DeviceInformation::FindAllAsync(DeviceClass::VideoCapture).get();
+        std::wofstream fileLogger("RTSPServer.log");
+        fileLogger << L"StartLogging..\n";
+        loggerConsole = StartLoggerConsole(L"RTSPServer.log");
+
+        if (!filteredDevices.Size())
+        {
+            throw_hresult(MF_E_NO_CAPTURE_DEVICES_AVAILABLE);
+        }
+        int selection = 1;
+        auto deviceIter = filteredDevices.First();
+        while (deviceIter.HasCurrent())
+        {
+            auto device = deviceIter.Current();
+            std::wcout << std::endl << selection++ << L". " << device.Name().c_str() << L":" << device.Id().c_str();
+            deviceIter.MoveNext();
+        }
+        std::cout << "\nEnter Selection:";
+        std::cin >> selection; selection--;

```

Comment:
nit: make this user input codes as separate functions for better code readability, GetDeviceSelectionFromUser or similar